Cython/Build/SharedModule.py

```diff
@@ -27,15 +26,23 @@ def generate_tree(compsrc):
 
         return generate_tree
 
+    def set_cimport_from_pyx(cimport_from_pyx):
+        def inner(node):
+            Options.cimport_from_pyx = cimport_from_pyx
+            return node
+        return inner
```

Comment:
Almost. We still need to make sure to reset the option to its original value in the end.

```suggestion

    orig_cimport_from_pyx = Options.cimport_from_pyx

    def set_cimport_from_pyx(cimport_from_pyx):
        def inner(node):
            Options.cimport_from_pyx = cimport_from_pyx
            return node
        return inner
```